packages/nextjs/test/performance/client.test.ts

```diff
@@ -28,28 +31,152 @@ describe('client', () => {
-      expect(mockStartTransaction).toHaveBeenCalledTimes(1);
-      expect(mockStartTransaction).toHaveBeenLastCalledWith({
-        name: '/[user]/posts/[id]',
-        op: 'pageload',
-        tags: {
-          'routing.instrumentation': 'next-router',
+    it.each([
+      [
+        'https://example.com/lforst/posts/1337?q=42',
+        '/[user]/posts/[id]',
+        { user: 'lforst', id: '1337', q: '42' },
+        {
+          _sentryGetInitialPropsTraceData: 'c82b8554881b4d28ad977de04a4fb40a-a755953cd3394d5f-1',
+          _sentryGetInitialPropsBaggage:
+            'other=vendor,foo=bar,third=party,last=item,sentry-release=2.1.0,sentry-environment=myEnv',
         },
-        metadata: {
-          source: 'route',
+        true,
+        {
+          name: '/[user]/posts/[id]',
+          op: 'pageload',
+          tags: {
+            'routing.instrumentation': 'next-router',
+          },
+          metadata: {
+            source: 'route',
+            baggage: [{ environment: 'myEnv', release: '2.1.0' }, '', true],
+          },
+          traceId: 'c82b8554881b4d28ad977de04a4fb40a',
+          parentSpanId: 'a755953cd3394d5f',
+          parentSampled: true,
         },
-      });
-    });
+      ],
+      [
+        'https://example.com/static',
+        '/static',
+        {},
+        {
+          pageProps: {
+            _sentryGetServerSidePropsTraceData: 'c82b8554881b4d28ad977de04a4fb40a-a755953cd3394d5f-1',
+            _sentryGetServerSidePropsBaggage:
+              'other=vendor,foo=bar,third=party,last=item,sentry-release=2.1.0,sentry-environment=myEnv',
+          },
+        },
+        true,
+        {
+          name: '/static',
+          op: 'pageload',
+          tags: {
+            'routing.instrumentation': 'next-router',
+          },
+          metadata: {
+            source: 'route',
+            baggage: [{ environment: 'myEnv', release: '2.1.0' }, '', true],
+          },
+          traceId: 'c82b8554881b4d28ad977de04a4fb40a',
+          parentSpanId: 'a755953cd3394d5f',
+          parentSampled: true,
+        },
+      ],
+      [
+        'https://example.com/',
+        '/',
+        {},
+        {},
+        true,
+        {
+          name: '/',
+          op: 'pageload',
+          tags: {
+            'routing.instrumentation': 'next-router',
+          },
+          metadata: {
+            source: 'route',
+          },
+        },
+      ],
+      [
+        'https://example.com/lforst/posts/1337?q=42',
+        '/',
+        {},
+        {},
+        false, // no __NEXT_DATA__ tag
+        {
+          name: '/lforst/posts/1337',
+          op: 'pageload',
+          tags: {
+            'routing.instrumentation': 'next-router',
+          },
+          metadata: {
+            source: 'url',
+          },
+        },
+      ],
+    ])(
+      'creates a pageload transaction (#%#)',
+      (url, route, query, props, hasNextData, expectedStartTransactionCall) => {
```

Comment:
```suggestion
      (url, route, query, props, hasNextData, expectedStartTransactionArguments) => {
```

It might be nice to put this list in a comment at the top of the `it.each`, and/or stick a blank line between test cases - it's pretty hard to tell what you're looking at given that it's half a dozen items per test case, and each test case goes on for a variable number of lines.